### PR TITLE
make a package out of coveo-testing/ref

### PR DESCRIPTION
**Pull request by jonapich**
The `*ref()` tool is so useful (and stable!) that I decided to make it a package on its own. It runs OOTB using the builtin python libraries too.

The `coveo-testing` package will replace its `ref` in a backwards-compatible manner:

- `ref` was defined in `mocks.py`
- it will import the symbols (they were not renamed) from coveo-ref and serve that
- users may decide to migrate to coveo-ref and drop coveo-testing if they only need ref

The code was not changed except:
- I wrote a new github action for the new package
- The pyproject.toml is new
- The exceptions were moved to a different file
- Imports and some tests were adjusted to fit the new package name

### CODE REVIEW

**Comment by dotboris** [comment]
> The coveo-testing package will replace its ref in a backwards-compatible manner

Do you mean in this PR or another one because this PR doesn't remove anything from `coveo-testing`

**Comment by jonapich** [comment]
> > The coveo-testing package will replace its ref in a backwards-compatible manner
> 
> Do you mean in this PR or another one because this PR doesn't remove anything from `coveo-testing`

In another PR. The workflows are not ordered so I need to be able to import `coveo-ref==1.0.0` to remove it cleanly.

File: coveo-ref/coveo_ref/__init__.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+    _bypass_context_check: bool = False,
+) -> Union[Tuple[str], Tuple[Any, str]]:
+    """
+    Replaces `resolves_mock_target`. Named for brevity.
+
+    Returns a tuple meant to be unpacked into the `mock.patch` or `mock.patch.object` functions in order to enable
+    refactorable mocks.
+
+    The idea is to provide the thing to mock as the target, and sometimes, the thing that is being tested
+    as the context. Refer to `coveo-testing`'s readme to better understand when a context is necessary.
```
Comment (pballandras) [inline coveo-ref/coveo_ref/__init__.py:332]: Wrong readme (I think)

**Review by pballandras — APPROVED** [review]

**Comment by jonapich** [comment]
@dotboris https://github.com/coveooss/coveo-python-oss/pull/171 that should do it

**Pull request merged**